src/group_theory/monoid_localization.lean

```diff
@@ -344,7 +345,10 @@ begin
   rw r_eq_r' at h ⊢,
   cases h with t ht,
   use t,
-  simp only [smul_mul_assoc, ht]
+  dsimp only [subtype.coe_mk] at ht ⊢,
+  haveI : smul_comm_class R M M :=
+    ⟨λ r m₁ m₂, by simp_rw [smul_eq_mul, mul_comm m₁, smul_mul_assoc]⟩,
```

Comment:
Could be replaced by this:
```suggestion
  rw [mul_comm ↑t, mul_comm ↑t, mul_comm b, mul_comm b'] at ht ⊢,
```
A bit weird that we suddenly need commutativity here; is this indicating we should but haven't flipped the direction somewhere?

Reply:
> A bit weird that we suddenly need commutativity here; is this indicating we should but haven't flipped the direction somewhere?

Ah, it seems that actually the right fix is to replace `is_scalar_tower R M M` by `smul_comm_class R M M`. Do you want me to go ahead and do that? Note this PR is getting dangerously close to the porting tide.

Reply:
Sorry what I suggested doesn't work as is: I should say the last three lines could be replaced by
```
  rw [mul_comm ↑t, mul_comm ↑t, mul_comm b, mul_comm b'] at ht ⊢,
  simp only [smul_mul_assoc, ht],
```
However, I agree that it's better to change the hypothesis to `smul_comm_class`, and this golf works against that, so I'm happy with the current proof. It's up to you whether to make the change; if not, feel free to merge after fixing the conflict.

Reply:
I think I'll just leave a TODO comment